### PR TITLE
feat(cdk): update tsconfig to more modern settings in typescript init templates 

### PR DESCRIPTION
**Pull request by ren-yamanashi**
### Issue # (if applicable)

n/A

### Reason for this change

I think contents of `tsconfig.json` are a little outdated and seemed to need update.

### Description of changes

The following two changes I made:

1. Removed unnecessary options
2. Updated `target` and `lib` options at compilerOptions

#### 1. Removed unnecessary options

Unnecessary options included in the `compilerOptions` of tsconfig.json.   
The unnecessary options are as follows

- noImplicitAny
- strictNullChecks
- noImplicitThis
- alwaysStrict

If `strict: true` is specified in compilerOptions, these options will also be `true`, so there is no need to specify them.

see: https://www.typescriptlang.org/tsconfig/#strict

#### 2. Updated `target` and `lib` options at compilerOptions

- target
  - I think 'ES2023' is more appropriate for the target option at this point.
- lib
  - Basically, it doesn't seem necessary to specify the same value for the lib option as the target specified in the target option.
  - see: https://www.typescriptlang.org/tsconfig/#lib

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by ren-yamanashi** [comment]
> Description of how you validated changes

If I update the contents of init-template, what steps should I take to verify the changes?

**Review by mrgrain — DISMISSED** [review]
Nice! Going by this https://www.npmjs.com/package/@tsconfig/node-lts we should set the target to `ES2022` and `lib` to include `ES2023`.

`strictPropertyInitialization` can also be removed.

**Comment by ren-yamanashi** [comment]
@mrgrain

> Nice! Going by this https://www.npmjs.com/package/@tsconfig/node-lts we should set the target to `ES2022` and `lib` to include `ES2023`.

Thank you review! I fixed in the following commit.

> `strictPropertyInitialization` can also be removed.

`strictPropertyInitialization` is currently set to `false`, but is it okay to remove it?   
Note: If `strict: true` is set, `strictPropertyInitialization` will implicitly become `true`.

If it is better not to remove, I will reset this commit. https://github.com/aws/aws-cdk/pull/31927/commits/6a66600011ba906133e3b5fc1ba031eb6ef2a28d

**Comment by mrgrain** [comment]
> > `strictPropertyInitialization` can also be removed.
> 
> `strictPropertyInitialization` is currently set to `false`, but is it okay to remove it?   
> Note: If `strict: true` is set, `strictPropertyInitialization` will implicitly become `true`.
> 
> If it is better not to remove, I will reset this commit. https://github.com/aws/aws-cdk/pull/31927/commits/6a66600011ba906133e3b5fc1ba031eb6ef2a28d

Ha nice catch. Let's keep it set to `false` then for now.

**Comment by ren-yamanashi** [comment]
> Ha nice catch. Let's keep it set to `false` then for now.

@mrgrain 

I have set `strictPropertyInitialization` back to false, so please check.

Also, there is no README and test, and CI seems to be failing. Should I address these issues?

The PR that made a similar fix to this PR did not seem to have updated the README or test...

https://github.com/aws/aws-cdk/pull/27422

**Review by mrgrain — APPROVED** [review]

**Comment by mrgrain** [comment]
> > Ha nice catch. Let's keep it set to `false` then for now.
> 
> @mrgrain
> 
> I have set `strictPropertyInitialization` back to false, so please check.
> 
> Also, there is no README and test, and CI seems to be failing. Should I address these issues?
> 
> The PR that made a similar change to this PR did not seem to have updated the README or test... #27422

Great! I can take care of the PR "requirements" ;) 

---

Disabled readme and test reqs. This is a change to code that has existing tests. Will run through the cli test pipeline.
